client/settings/advanced-settings-section/experimental-features.js

```diff
@@ -0,0 +1,125 @@
+import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { CheckboxControl, ExternalLink } from '@wordpress/components';
+import { getQuery, updateQueryString } from '@woocommerce/navigation';
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+import { useGetSavingError, useIsUpeEnabled, useSettings } from '../../data';
+import UpeToggleContext from '../upe-toggle/context';
+import { STORE_NAME } from '../../data/constants';
+
+const ExperimentalFeatures = () => {
+	const dispatch = useDispatch();
+	const [ isUpeEnabled, setIsUpeEnabled ] = useIsUpeEnabled();
+	const { isSaving } = useSettings();
+	const savingError = useGetSavingError();
+	const { setIsUpeEnabledLocally } = useContext( UpeToggleContext );
+	const isUpeEnabledBeforeSaving = useRef( isUpeEnabled );
+	const headingRef = useRef( null );
+	const setIsUpeEnabledBeforeSaving = useCallback( ( value ) => {
+		isUpeEnabledBeforeSaving.current = value;
+	}, [] );
+	const hasSavedSettings = useRef( false );
+	const setHasSavedSettings = useCallback( ( value ) => {
+		hasSavedSettings.current = value;
+	}, [] );
+
+	useEffect( () => {
+		if ( ! headingRef.current ) {
+			return;
+		}
+
+		const { highlight } = getQuery();
+		if ( highlight === 'enable-upe' ) {
+			headingRef.current.focus();
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( isSaving && ! hasSavedSettings.current ) {
+			setHasSavedSettings( true );
+		} else if ( ! isSaving && hasSavedSettings.current && ! savingError ) {
+			if ( isUpeEnabled !== isUpeEnabledBeforeSaving.current ) {
+				if ( isUpeEnabled ) {
+					dispatch( 'core/notices' ).createSuccessNotice(
+						__(
+							'✅ New checkout experience enabled.',
+							'woocommerce-gateway-stripe'
+						),
+						{
+							actions: [
+								{
+									label: __(
+										'Review accepted payment methods',
+										'woocommerce-gateway-stripe'
+									),
+									onClick: () => {
+										updateQueryString(
+											{ panel: 'methods' },
+											'/',
+											getQuery()
+										);
+										// It doesn't seem to be possible to programatically switch the tab.
+										// For that reason, we need to reload so the user can review their PMs.
+										window.location.reload();
+									},
```

Comment:
I couldn't find an API to switch the tab in the `TabPanel` component, that would be infinitely better than reloading. Perhaps @ricardo or @diegocurbelo could indicate if there's an API/feasible approach to achieve this?

Reply:
I think there isn't a clean way of doing the tab switch. This is a feature we can suggest in the Guttenberg repo.

Reply:
There's already a feature request for it [here](https://github.com/WordPress/gutenberg/issues/27371).